[test] Test test_ad_basic.py failed on MacOS Mojave

<!-- We've collected some common issue solutions in https://taichi.readthedocs.io/en/stable/install.html#troubleshooting. Make sure you've check them out first. Hopefully they could address your problem. -->

**Describe the bug**
I have a developement build on my MacOS, and didn't manage to pass the test of `test_ad_basic.py`.

Logs shown as below:

**To Reproduce**
`pytest test_ad_basic.py`

**Logs**
```
========================= FAILURES ==========================
____________________ test_trigonometric _____________________

args = (), kwargs = {}

    def wrapper(*args, **kwargs):
        if has_autograd:
>           func(*args, **kwargs)

test_ad_basics.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_ad_basics.py:80: in test_trigonometric
    grad_test(lambda x: ti.sin(x), lambda x: np.sin(x))
test_ad_basics.py:50: in grad_test
    impl()
../../python/taichi/lang/__init__.py:449: in wrapped
    test(*test_args, **test_kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    @ti.all_archs_with(default_fp=default_fp)
    def impl():
        print(f'arch={ti.cfg.arch} default_fp={ti.cfg.default_fp}')
        x = ti.var(default_fp)
        y = ti.var(default_fp)
    
        ti.root.dense(ti.i, 1).place(x, x.grad, y, y.grad)
    
        @ti.kernel
        def func():
            for i in x:
                y[i] = tifunc(x[i])
    
        v = 0.234
    
        y.grad[0] = 1
        x[0] = v
        func()
        func.grad()
    
>       assert y[0] == approx(npfunc(v))
E       assert 0.23188334703445435 == 0.2318703549116868 ± 2.3e-07
E        +  where 0.2318703549116868 ± 2.3e-07 = approx(0.2318703549116868)
E        +    where 0.2318703549116868 = <function test_trigonometric.<locals>.<lambda> at 0x123d80378>(0.234)

test_ad_basics.py:47: AssertionError
------------------- Captured stdout call --------------------
Running test on arch=Arch.x64
[Taichi] Starting on arch=x64
arch=Arch.x64 default_fp=DataType.float32
[Taichi] materializing...
Running test on arch=Arch.metal
[Taichi] Starting on arch=metal
arch=Arch.metal default_fp=DataType.float32
[Taichi] materializing...
Running test on arch=Arch.x64
[Taichi] Starting on arch=x64
arch=Arch.x64 default_fp=DataType.float32
[Taichi] materializing...
Running test on arch=Arch.metal
[Taichi] Starting on arch=metal
arch=Arch.metal default_fp=DataType.float32
[Taichi] materializing...
```

--- a/python/taichi/testing.py
+++ b/python/taichi/testing.py
@@ -1,7 +1,7 @@
 import taichi as ti
 
 
-## Helper functions
+# Helper functions
 def approx(expected, **kwargs):
     '''Tweaked pytest.approx for OpenGL low percisions'''
     import pytest
@@ -22,6 +22,9 @@ def approx(expected, **kwargs):
     if ti.cfg.arch == ti.opengl:
         kwargs['rel'] = max(kwargs.get('rel', 1e-6), 1e-3)
 
+    if ti.cfg.arch == ti.metal:
+        kwargs['rel'] = max(kwargs.get('rel', 1e-6), 1e-4)
+
     return pytest.approx(expected, **kwargs)
 
 
@@ -39,7 +42,7 @@ def make_temp_file(*args, **kwargs):
     return name
 
 
-## Pytest options
+# Pytest options
 def _get_taichi_archs_fixture():
     import pytest
 

